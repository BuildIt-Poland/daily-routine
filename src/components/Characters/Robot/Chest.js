import React from 'react';
import styled from 'styled-components';

import { zIndexArtworkPart } from '../../../styles/designTokens';

function ChestArtwork(props) {
  return (
    <svg xmlns="http://www.w3.org/2000/svg" width="117" height="93" viewBox="0 0 117 93" {...props}>
      <path fill="currentColor" d="M13.287 5.033L4.614 88.741l109.756-5.08-10.789-81.852z" />
      <path
        d="M80.341 43.583c-.567.867-1.404 4.231-1.883 5.154l-34.545 3.415-1.883-31.457c.158-.311.253-.678.253-1.108v-7.25c.448-.514.893-1.029 1.312-1.565l28.359-.604 8.221 28.589c-.2.437-.269.956-.136 1.542.244 1.084.253 2.186.302 3.284z"
        fill="#fff"
      />
      <path
        d="M115.591 78.932c-.145-1.465-.558-3.353-.782-4.805l-1.227-7.86c-.065-.605-.295-2.094-.372-2.697-.302-2.36-.69-4.331-.967-6.697-.088-.754-.302-2.594-.382-3.348l-.651-5.255c-.19-1.988-.596-4.247-.808-6.232-.253-2.369-.559-4.89-1.116-7.209-.553-2.302-.488-3.535-.837-5.86-.03-.203-.274-.703-.203-.858-.019-.205-.031-.417-.059-.628a1234.192 1234.192 0 0 0-1.716-12.735c-.153-1.087-.404-2.194-.427-3.292-.012-.563-.427-1.78-.151-2.241-.089.15.1.58.138.74-.479-2.859-.947-5.754-1.24-8.638-.024-.235.047-.723-.147-.903-.18-.166-.79-.032-.995-.02-1.179.07-2.363.092-3.544.135-5.155.187-10.321.182-15.48.266 0 0-2.997.107-3.181.054-3.374-.051-8.021.145-11.778.223-4.307.089-8.614.189-12.92.301.123.017.246.035.368.055-6.61.187-24.288.88-24.465.887-.535.022-1.07.044-1.606.059-3.885.111-7.777.032-11.664.06l-5.55.04c-.64.005-1.514-.18-2.139-.044-.594.129-.538.712-.549 1.235-.078 3.533-.447 7.076-.695 10.601-.068.966-1.084 12.417-1.12 12.932-.252 2.842-.23 7.58-.237 7.52-.017 1.289-.071 5.843-.322 5.915-.049.014-.148-.249-.173-.27-.033-.026-.361 2.235-.367 2.406-.073 1.964-.449 3.927-.32 5.896-.041-.217-.427-3.331-.433-4.021-.118 1.115.184 4.015-.467 4.574a.514.514 0 0 1-.088-.207c-.467.692-.292 2.042-.309 2.827-.03 1.418-.184 2.838-.219 4.248-.019.787-.942 8.161-.715 8.162a1.22 1.22 0 0 0 .303-.685l-.217 1.858c-.011-.047-.062-.332-.16-.36-.249-.07-.736 3.282-1.011 1.892-.418 1.343-.092 2.461-.034 3.741.01.222-.664 3.584-.711 3.548a.526.526 0 0 0-.219-.262c.028 2.04-.716 4.049-.791 6.122-.027.757.094 1.528.105 2.249.004.296-.29 2.731-.643 2.293-.081.866.737-8.591 1.338-14.178C2.32 77.825 1.383 85.14.672 92.575c1.506.31 3.578-.233 5.124-.332 1.556-.099 3.116-.175 4.669-.31a58.576 58.576 0 0 0 4.515-.579 46.612 46.612 0 0 0 2.987-.587c.751-.175 1.808-.568 2.517-.516.383.028 2.099.341 2.486.427-.044-.1-.086-.201-.13-.3.147.031.303.074.482.127 1.223.365 2.527.457 3.797.496.669.02 1.338.019 2.007.019.175 0 .582-.053.963-.074a.943.943 0 0 1 .385-.121c4.497-.437 9.007-.699 13.502-1.164 2.896-.299 5.764-.687 8.649-.817l-.002-.005c1.937-.159 3.874-.308 5.813-.442l.564-.038a.93.93 0 0 1 .256-.107c2.207-.538 4.439-.351 6.688-.342 2.79.011 5.594-.128 8.38-.268 3.812-.191 7.588-.499 11.382-.666l-.017-.048c6.396-.319 27.518-1.447 30.642-1.605a165.42 165.42 0 0 1-.74-6.391zm-4.28 2.988c-2.393.179-4.789.317-7.186.407-5.065.19-10.136.179-15.204.21-5.412.033-10.758.208-16.148.72-1.376.131-2.782.374-4.163.402-1.157.023-2.318-.175-3.472-.252-.741-.05-1.533-.151-2.273-.047-.935.131-.038.33.423.33-2.948.188-5.901.279-8.853.392-2.899.112-5.849.602-8.745.472-1.424-.064-2.831-.004-4.255-.009-1.434-.005-2.872-.128-4.304-.071-2.931.117-5.867.951-8.799.839l.592-.067c-2.122.069-4.244-.048-6.364-.13-.588-.023-1.177-.123-1.765-.136-.044-.113-4.028-.41-5.982-.349-2.053.064-4.113.217-6.164.254.283-6.527 1.02-13.123 1.565-19.643-.22-.085-.147 2.301-.553 2.375.508-6.122 1.083-12.295 1.824-18.392.196-1.613.355-3.303.565-4.883.244-1.831.758-4.004.792-6.059.041-2.453.435-4.72.309-7.245-.05-.989-.098-1.978-.146-2.967l1.103-11.525c.063-.3.047-.689.081-.957.195-1.525.389-3.05.586-4.574.164-1.278.387-2.564.457-3.85.018-.331-.035-.857.262-1.083.275-.209.866-.092 1.195-.103 3.233-.11 6.472-.422 9.7-.633l21.536-1.405c-.511-.095-1.023-.187-1.534-.279 13.862-.107 27.723-.306 41.583-.563 2.781-.051 5.562-.117 8.342-.188 1.541-.039 3.082-.067 4.624-.096.371-.007 1.235-.216 1.548-.001.364.251.323 1.391.373 1.763.445 3.31.958 6.61 1.391 9.921.446 3.407.73 6.833 1.154 10.243.965 7.766 3.057 24.098 3.13 24.726l.724 6.636.003-.001c.104.902.213 1.804.337 2.704.344 2.51.587 5.031.897 7.545 0 0 .109.895.15 1.201.052.388.105.777.159 1.165.024.171.156 1.004.162 1.008l.217 1.535-.018-.001c.151 1.048.487 3.32.487 3.32h.001c.262 1.768.523 3.537.768 5.307.07.504.151 1.009.195 1.516.065.753-.638.468-1.307.518zM84.677 42.629c-.314-1.157-.521-2.343-.777-3.514-.714-3.273-1.74-6.485-2.412-9.762-.065-.319-.035-.515-.279-.682-.106-.072-.749-.037-.901-.037-.626 0-1.256.073-1.883.083-1.679.028-3.357.078-5.034.145-1.576.062-3.133-.022-4.708-.025-.818-.002-1.637.006-2.456.014-.41.004-.82.011-1.229.018-.007 0-.85.01-.5.02-1.457.026-2.913-.08-4.37-.048-1.472.032-2.942.136-4.414.189.255 2.497.345 5 .348 7.51.002 1.208.11 2.33.365 3.511.089.415.139.774.091 1.192-.031.275-.275.819-.026 1.054.037.035.078-.201.078-.202.276 2.311.191 4.625.408 6.935.542-.074 1.071-.217 1.606-.327.244-.05.489-.1.737-.129.08-.01.851-.103.446-.008.901-.084 1.809.008 2.713-.057.836-.061 1.67-.161 2.502-.255 1.693-.191 3.383-.41 5.078-.576.008-.086-.528.068-.567-.063 2.025-.201 4.063-.171 6.092-.314.711-.05 1.457-.071 2.163-.211a4.602 4.602 0 0 1 1.743-.736c-.064-.005-.128-.007-.192-.013.249-.073.502-.084.757-.079.758-.098 1.555-.112 2.438-.252.537-.085 1.065-.175 1.589-.259.025-.161.211-.532.209-.537a.37.37 0 0 0-.289-.027c.067-.009.092.007.076.048-.436.237-1.32.094-1.793.123-.234.015-.46.055-.688.094-.28.136-.605.217-.976.219-.702.004-1.403.014-2.105-.014-.275-.011-.55-.03-.826-.03-.199.001-.429.047-.428.022-.064.011-.129.021-.191.033a3.01 3.01 0 0 1-.525.098c-1.004.191-1.837-.092-2.873.269.075-.041.918-.353.913-.373-.339-.002-.879.075-1.048.052-.426-.06-.785.014-1.215.108-.39.085-.716.107-1.115.106-.251 0-.529-.03-.776.026-.013.003-.619.209-.302.236l-.359.023c.229-.136-.741-.219-.41-.301-.404-.081-.775.226-1.169.222-.169-.002-.244-.044-.395-.062a7.767 7.767 0 0 0-.804-.057c-.433-.003-.773.099-1.183.151-.383.048-.42-.081-.774-.053a82.681 82.681 0 0 1 3.48-.324c-2.15-.047-4.399-.107-6.477.536-.413.128-.749.311-1.18.288-.319-.017-.622-.092-.954-.092-.189 0-.704.081-.88-.026-.291-.178-.252-.694-.28-1.002-.093-1.027-.219-2.05-.292-3.08-.037-.519-.187-1.207-.022-1.688.145-.422.499-.852.74-1.226.68-1.057 1.837-2.989 3.36-2.654 1.026.226 1.608 1.278 2.11 2.102.604.992 1.316 1.914 2.23 2.644 1.835 1.466 4.158 1.591 5.732-.296.73-.874 1.11-1.913 1.403-2.997.314-1.161.661-2.395 1.76-3.057 2.257-1.36 5.016.807 6.321 2.544.438.583.588 1.348.845 2.028.266.704.621 1.365.897 2.065.29.736.457 1.541.779 2.26.033.074.475.716.411.781.371-.381-.23-1.885-.343-2.304zm-8.753-8.091c-1.981.74-2.641 2.934-3.475 4.657-.474.98-1.202 2.096-2.387 2.231-1.121.128-2.244-.649-2.976-1.427-1.169-1.243-1.852-3.046-3.363-3.948-2.373-1.416-4.448 1.175-5.444 3.046l-.04-1.066c-.059-1.705-.061-3.429-.256-5.125-.042-.37-.09-.743-.078-1.116.017-.507-.015-.643.499-.58.814.099 1.636.054 2.453.088l-1.031-.07c1.202-.097 2.404-.202 3.609-.273-.236-.066-.468-.148-.711-.188 1.192.077 2.423-.11 3.629-.072-.453-.162.401-.046.522-.045.34.004.68.004 1.02.004.736 0 1.473-.007 2.209-.002.655.005 1.29.006 1.943-.031.223-.013.415.003.629.043.069.013.555.107.305.118 1.275-.177 2.566-.189 3.851-.201.614-.006 1.227-.008 1.841-.019.259-.005.944-.161 1.171 0 .3.213.394 1.014.488 1.325.194.645.384 1.29.578 1.934.386 1.281.68 2.033 1.102 3.328-1.356-1.695-3.805-3.463-6.088-2.611zm-1.201-11.823c4.745-.664 5.255-7.24 1.791-9.771-1.77-1.293-4.21-1.242-5.859.236-1.54 1.379-2.121 3.666-1.55 5.634.695 2.397 3.068 4.257 5.618 3.901zm-1.229-9.066c2.182-.562 4.386 1.09 4.8 3.237.389 2.019-.833 4.266-2.871 4.79-2.196.564-4.293-1.132-4.713-3.254-.399-2.018.722-4.242 2.784-4.773zm-13.393 9.186c4.692-1.047 5.393-7.777 1.424-10.144-1.841-1.098-4.357-.926-5.857.678-1.438 1.537-1.589 3.842-.95 5.763.775 2.33 2.767 4.287 5.383 3.703zm-4.399-4.181c-.568-2.007.903-4.369 2.859-4.952 2.217-.661 4.446 1.193 4.828 3.34.351 1.969-.656 4.2-2.778 4.55-1.929.318-4.362-1.005-4.909-2.938zm-4.43 61.072c-.007.131-.001.261.017.39a23.335 23.335 0 0 1-.409-4.844l.08.495c.002-.114.085-.479.081-.505.387.967.403 1.861.499 2.898.067.722.211 1.465.205 2.19 0-.014.061-.368.031-.408.056.155.169.469.111.632.033-.271.046.545.04.638-.007-.093.024-.666-.126-.738-.007.115-.008.23-.004.346-.059-.062-.413-.62-.425-.212l-.1-.882zm-.568-7.621c-.085.08-.327-1.468-.333-1.517-.124-.964-.268-1.926-.398-2.89-.124-.914-.24-1.83-.318-2.749-.073-.857.022-1.724-.08-2.575-.464-3.89-1.08-7.741-1.39-11.652.202.137.055-.707.044-.892-.025-.426-.058-.853-.074-1.279-.028-.737.047-1.471.031-2.207-.032-1.53-.462-3.016-.645-4.529-.385-3.18-.38-6.407-.661-9.598-.28-3.182-.843-6.331-1.083-9.518a28.484 28.484 0 0 1-.01-4.213c.012-.163.217-1.86.193-1.87.148.059.149.591.151.358.002.193.159.929.363.751.172-.15-.071-.982-.11-1.172-.203-1.01-.847-2.38-.291-3.346.505.207.034 1.056.438 1.363-.086-1.994-.728-4.351-.181-6.305.43 2.032.395 4.188.626 6.251.06.538.12 1.076.168 1.616.02.219-.007.52.058.727.049.155.15.241.189.354.301.87.167 1.897.359 2.801-.229-.333.163.768.193.913-.641-.259-.574 1.001-.552 1.331a.37.37 0 0 0-.153-.316c.063.173.079.349.048.53-.009-.064-.102-.549-.189-.509-.051.023-.122.362-.076.393-.275-.186.003-.733-.203-.982.092 1.531.224 3.077.645 4.558l-.031-.976c.482 2.721.813 5.465 1.247 8.194l-.332-.224c.626 6.789 1.073 13.593 1.632 20.388.144 1.746.295 3.492.464 5.236.088.911.043 1.856.202 2.757.06.343.187.609.237.93.121.784.149 1.671.181 2.466.03.739-.049 1.504-.125 2.238-.086.828-.016 1.589-.108 2.414l-.023-.513c-.796.224.297 2.886-.103 3.263z"
        fill="#222"
      />
    </svg>
  );
}

const Chest = styled(ChestArtwork)`
  position: absolute;
  top: 6.4rem;
  left: 1.1rem;
  z-index: ${zIndexArtworkPart + 3};
`;

export default Chest;
