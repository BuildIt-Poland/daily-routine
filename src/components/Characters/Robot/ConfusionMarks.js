import styled from 'styled-components';

import ConfusionMarksBase from '../ConfusionMarks';

const ConfusionMarks = styled(ConfusionMarksBase)`
  top: 4.8rem;
  left: 6rem;
`;

export default ConfusionMarks;
