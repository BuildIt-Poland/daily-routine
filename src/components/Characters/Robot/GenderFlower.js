import styled from 'styled-components';

import GenderFlowerBase from '../GenderFlower';

const GenderFlower = styled(GenderFlowerBase)`
  top: -1.6rem;
  left: 3rem;
`;

export default GenderFlower;
