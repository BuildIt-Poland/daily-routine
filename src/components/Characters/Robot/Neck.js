import React from 'react';
import styled, { keyframes } from 'styled-components';

import { zIndexDefault } from '../../../styles/designTokens';

const neckAnimation = keyframes`
  0% { transform: translateX(0) rotate(-5deg); }
  50% { transform: translateX(0) rotate(10deg); }
  100% { transform: translateX(0) rotate(-5deg); }
`;

const Neck = styled(NeckArtwork)`
  animation: ${neckAnimation} 110ms ease-in-out infinite;
  position: absolute;
  top: 5.6rem;
  left: 4.4rem;
  z-index: ${zIndexDefault};
`;

function NeckArtwork(props) {
  return (
    <svg xmlns="http://www.w3.org/2000/svg" width="52" height="16" viewBox="0 0 52 16" {...props}>
      <path fill="currentColor" d="M2.326 0h47.795v16H2.326z" />
      <path d="M.412 16L.956 0H4.24v16H.412zm51.176 0V0h-2.933v16h2.933z" fill="#222" />
    </svg>
  );
}

export default Neck;
