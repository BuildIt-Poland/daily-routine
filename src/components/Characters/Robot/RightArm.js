import React from 'react';
import styled, { keyframes } from 'styled-components';

import { zIndexArtworkPart } from '../../../styles/designTokens';

const rightArmAnimation = keyframes`
  0% { transform: translateX(0) rotate(0deg); }
  50% { transform: translateX(0) rotate(-2deg); }
  100% { transform: translateX(0) rotate(0deg); }
`;

const RightArm = styled(RightArmArtwork)`
  animation: ${rightArmAnimation} 130ms ease-in-out -0.2s infinite;
  transform-origin: 3.6rem 1rem;
  position: absolute;
  top: 7.4rem;
  left: 0.3rem;
  z-index: ${zIndexArtworkPart + 4};
`;

function RightArmArtwork(props) {
  return (
    <svg xmlns="http://www.w3.org/2000/svg" width="53" height="119" viewBox="0 0 53 119" {...props}>
      <path
        fill="currentColor"
        d="M29.846 84.136s-5.047-6.187-7.953-16.975c-1.407-5.223-2.764-11.574-1.941-18.731 2.487-21.622 17.698-25.417 20.736-26.431 2.914-4.424 4.986-21.204-9.152-19.778 0 0-4.906 1.56-8.973 4.64-1.175.889-1.01 1.809-.646 2.518-1.263 1.566-4.763 3.723-5.697 4.375C6.546 20.514.833 39.668.957 45.993c.025 1.298-.09 3.2-.121 5.597.17.1.313.26.389.496.699 2.167.811 3.978.776 6.238-.035 2.313 1.786 11.791 2.032 13.706C6.3 79.484 9.876 86.875 17.459 95.669c0 0-3.907 15.161 11.1 21.75 0 0-4.961-13.688 4.589-18.773 9.549-5.085 15.083-2.412 18.079.566-.531-4.287-5.073-18.177-21.381-15.076z"
      />
      <path
        fill="#222"
        d="M32.525 22.737c.414 1.136 1.014 2.492-.104 3.384-.964.769-1.982 1.396-2.897 2.241a27.465 27.465 0 0 0-5.033 6.256c-1.179-1.507-2.514-2.88-3.806-4.289-.616-.672-1.15-1.447-1.974-1.87-.777-.399-1.456-.905-2.195-1.372-2.28-1.442-4.734-2.842-7.457-3.107 1.067-.912.879-2.771 2.163-3.479-.026.374-1.12 1.537-.673 1.726.22.093.82-.559.959-.718.245-.281.568-.708.482-1.108-.274.095-.544.041-.823.031.109-.414.211-.973.502-1.303.11-.124.294-.355.494-.238.269.156-.291.6-.405.786.24.321.906-.522 1.094-.636-.699-.027-.597-1.129-1.402-.672.653-.876 1.295-1.713 2.146-2.406.143-.117.361-.218.45-.387.088-.167.016-.327.064-.483.135-.433.641-.739.929-1.067-.077.055-.464.236-.473.319.066-.662 1.385-1.436 1.852-1.816.331-.27.776-.558 1.152-.889a.825.825 0 0 0 .352-.254c.973-1.161 2.498-1.514 3.883-1.955.04-.013.074-.031.11-.048.402.19.803.456 1.163.661 4.773 2.718 7.618 7.675 9.447 12.693zM4.647 73.988c-1.435-4.463-2.688-8.98-3.042-13.673-.178-2.359-.387-4.717-.551-7.077-.045-.65-.368-1.921-.011-2.532.209-.357.658-.402 1.021-.501.563-.154 1.119-.329 1.68-.49-.049.792.006 1.585-.015 2.377.068-.765-.01-2.258.761-2.63.451-.217 1.063-.254 1.547-.381.655-.172 1.265-.283 1.943-.274 1.619.022 3.231.215 4.839.384 1.631.171 3.079.151 4.64.704 1.182.419 2.102.684 2.244 2.12.148 1.503.018 3.05.135 4.562a45.927 45.927 0 0 0 1.768 9.456c-2.286.262-4.531.779-6.742 1.404-.976.276-2.253.437-3.136.936-.393.222-.72.56-1.119.789-.484.276-.991.513-1.448.834-1.659 1.166-2.732 3.028-4.514 3.992zm-.131-18.565c.306-.801.066-1.728.146-2.721-.204.52-.221 1.739-.146 2.721zm-.268-2.058c.089-.28.233-.58.218-.881a.93.93 0 0 1-.14-.315c-.016.397-.02.803-.078 1.196zm-.381 2.428l-.286-1.449c-.121.591.054.899.286 1.449zm.936 4.538c-.09-.974-.517-1.951-.271-2.936-.076.445-.232.266-.376.434-.223.259-.255.602-.253.957a.635.635 0 0 1 .156-.334c-.025.01.137 1.36.145 1.468-.004-.097-.069-.693.099-.733.25-.059.277 1.202.5 1.144zm46.501 39.885c-1.586-.929-4.86-2.627-7.646-2.654-.037 0-.07.006-.104.009-.676-.032-1.658.188-1.978.211-.498.036-1.016-.018-1.511.053-.132.019-.478.196-.587.112-.006-.075.03-.133.109-.173-1.301.274-2.529.342-3.759.891-.974.435-2.097.841-2.719 1.767.068-.128.144-.252.227-.371-1.389 1.168-3.053 2.74-3.576 4.52-.65 2.214-1.083 4.544-.733 6.85.153 1.008.42 2.035.846 2.965.398.866 1.115 1.735 1.021 2.746-.066.717-.688 1.419-1.416 1.502-.802.091-1.833-.513-2.56-.812-6.791-2.801-11.178-9.912-10.488-17.25.096-1.024.411-1.983.623-2.983.191-.907-.208-1.039-.771-1.673-1.322-1.486-2.497-3.221-3.553-4.903-.521-.829-.942-1.741-1.564-2.5-.524-.64-1.013-1.197-1.358-1.96-.738-1.633-1.974-2.981-2.545-4.7-.301-.908-.425-1.866-.724-2.778-.246-.751-.723-1.472-.505-2.286.397-1.482 1.729-3.068 2.793-4.127 2.68-2.666 6.549-3.734 10.164-4.398.781-.143 1.606-.441 2.4-.283.747.148.926.277 1.177.977a68 68 0 0 0 1.718 4.313c.157.357.306.989.579 1.284.169.183.228.213.376.21.347.929.894 1.773 1.365 2.654.329.616.709 1.197 1.046 1.807.238.43.403.886.6 1.328l-.266.054c.356.623.714 1.243 1.085 1.856.147.242.562 1.223.836 1.284.253.056 1.029-.407 1.297-.482.617-.172 1.275-.234 1.914-.252 2.497-.07 5.109.439 7.494 1.148l-.015-.039c1.562.598 2.997 1.224 4.332 2.051l-.176.005c.603.391 1.229 1.057 1.901 1.231.443.368.876.769 1.299 1.214a.921.921 0 0 0 .231.175c.258.432.618.788.936 1.255.634.932 1.481 2.571 1.747 3.633.283 1.126.673 2.129.953 3.255.179.718.25 2.031.344 3.039a9.359 9.359 0 0 1-.859.225zm-2.25-3.341a15.602 15.602 0 0 0-3.854-6.368c-3.195-3.208-7.7-4.993-12.228-4.81-2.65.107-5.463.832-7.645 2.387-.967.69-2.234 1.612-2.888 2.616-.302.464-.436 1.013-.757 1.469-.326.465-.779.826-1.098 1.297.003-.115.007-.229.012-.344-.191.339-.435.652-.598 1.008-.11.24-.118.347-.081.459-.687 1.198-1.179 2.577-1.234 3.833-.046 1.036.074 2.067.023 3.104-.038.77.097 1.311.313 1.911.047.328.107.657.185.988.27 1.143.878 2.064 1.323 3.135.494 1.189 1.2 2.271 2.039 3.245.798.927 1.679 1.836 2.724 2.486.222.138.715.508.929.171.221-.349-.492-.715-.267-.884-.88-.803-.23-1.823.17-2.681.493-1.057.459-2.127.626-3.257.341-2.297 1.551-4.56 3.006-6.34 2.701-3.304 6.814-5.398 11.085-5.573a14.631 14.631 0 0 1 8.215 2.148zM.867 48.306c.013-2.636.35-5.257.796-7.851.209-1.213.495-2.362.846-3.506.124-.404.236-.82.374-1.22.127-.367.28-.443.488-.711.115-.149.113-.382.246-.526.119-.13.324.003.497-.115.145-.099.22-.391.281-.534.176-.411.55-1.047.55-1.497 0-.138-.146-.361-.13-.463.011-.071.167-.169.188-.259.065-.282.111-.39-.065-.618-.188-.242-.566-.169-.239-.692.226-.363.387.018.669-.264.236-.235.264-.922.362-1.232.204-.643.572-2.062 1.263-2.333.27-.106.79-.028.86.253.074.293-.348.806-.506 1.031.526.47.998-.725 1.001-1.102.763.128 1.516.305 2.278.437.475.082.952.089 1.391.286.352.158.674.362 1.022.524.577.269 1.195.189 1.785.402 1.141.413 2.261 1.405 3.203 2.149a27.427 27.427 0 0 1 3.041 2.822c.498.534.965 1.093 1.43 1.656.361.438.843.919.81 1.532-.032.596-.513 1.157-.766 1.674a16.2 16.2 0 0 0-.781 1.942c-.967 2.848-1.569 5.82-1.86 8.812-3.298-1.362-6.82-2.326-10.406-2.396-3.058-.059-5.926.268-8.628 1.799zm8.558-14.35c.461-.781 1.17-1.443 1.393-2.349-.257.265-.589.449-.787.769-.299.482-.411 1.054-.606 1.58zm-4.547.018c.326.065.63-.723.173-.89-.07.294-.116.593-.173.89zm-.179 5.037c.777-2.115 1.789-4.228 2.806-6.237-.955 1.774-3.553 3.988-2.806 6.237zm3.228-4.007c-1.221 2.335-2.934 4.551-3.188 7.259.315.008 1.173-1.3.978-.033.795-2.388 1.765-4.736 2.21-7.226zM35.919.449c-2.435-.519-4.651 1.086-6.699 2.179a70.942 70.942 0 0 0-3.495 1.971c-.528.321-1.051.653-1.564.997-.325.218-1.636.946-1.599 1.264.042.364.806.619 1.072.753.529.268 1.036.579 1.528.91a18.799 18.799 0 0 1 2.98 2.5c1.808 1.86 3.237 4.059 4.369 6.386a35.018 35.018 0 0 1 1.576 3.82c.207.594.399 1.193.579 1.795.075.25.13.899.302 1.103.271.322.728-.096 1.111-.269a48.758 48.758 0 0 1 1.873-.788c1.177-.472 2.48-.839 3.558-1.519 3.522-2.222 4.231-7.247 3.48-11.01-.933-4.668-4.235-9.061-9.071-10.092z"
      />
    </svg>
  );
}

export default RightArm;
