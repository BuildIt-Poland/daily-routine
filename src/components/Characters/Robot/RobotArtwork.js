import React from 'react';
import styled from 'styled-components';

import { pose } from '../../../types';
import { DEFAULT } from '../../../constants/roleActions';
import Head from './Head';
import Neck from './Neck';
import Chest from './Chest';
import Shadow from './Shadow';
import RighArm from './RightArm';
import LeftArm from './LeftArm';
import RightLeg from './RightLeg';
import LeftLeg from './LeftLeg';

const Frame = styled.div`
  position: relative;
  height: 24.4rem;
  width: 14.8rem;
  color: ${props => props.theme.primaryColor};
`;

function RobotArtwork({ pose = DEFAULT }) {
  return (
    <Frame>
      <Head pose={pose} />
      <Neck />
      <Chest />
      <RighArm />
      <LeftArm />
      <RightLeg />
      <LeftLeg />
      <Shadow />
    </Frame>
  );
}

RobotArtwork.propTypes = {
  pose
};

export default RobotArtwork;
