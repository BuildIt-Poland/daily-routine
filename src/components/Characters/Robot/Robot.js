import React from 'react';
import styled from 'styled-components';

import { MALE, FEMALE } from '../../../constants/genders';
import { CONFUSED } from '../../../constants/characterMoods';
import { gender, mood } from '../../../types';

import { spacingLarge } from '../../../styles/designTokens';
import RobotArtwork from './RobotArtwork';
import ConfusionMarks from './ConfusionMarks';
import GenderFlower from './GenderFlower';

const Wrapper = styled.div`
  margin: 7.6rem 0 ${spacingLarge} 0;
  position: relative;
`;

function Robot({ gender = MALE, mood = CONFUSED }) {
  return (
    <Wrapper>
      <RobotArtwork mood={mood} />
      {mood === CONFUSED && <ConfusionMarks />}
      {gender === FEMALE && <GenderFlower />}
    </Wrapper>
  );
}

Robot.propTypes = {
  gender,
  mood
};

export default Robot;
