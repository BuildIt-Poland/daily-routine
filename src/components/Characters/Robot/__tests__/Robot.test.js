import React from 'react';
import { create } from 'react-test-renderer';

import { DEFAULT } from '../../../../constants/roleActions';
import Robot from '../Robot';

jest.mock('../ConfusionMarks', () => 'ConfusionMarks');
jest.mock('../RobotArtwork', () => 'RobotArtwork');

describe('COMPONENT - Characters Robot', () => {
  it('renders correctly with default props', () => {
    const component = create(<Robot />);

    expect(component.toJSON()).toMatchSnapshot();
  });

  it('renders confusion marks if robot is in default pose', () => {
    const component = create(<Robot pose={DEFAULT} />);

    expect(component.toJSON()).toMatchSnapshot();
  });
});
