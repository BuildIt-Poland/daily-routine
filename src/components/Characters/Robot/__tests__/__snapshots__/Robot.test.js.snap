// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENT - Characters Robot renders confusion marks and flower if female robot is in default pose 1`] = `
.c0 {
  padding: 9.6rem 0 3rem 0;
  position: relative;
}

<div
  className="c0"
>
  <RobotArtwork
    pose="default"
  />
  <ConfusionMarks />
  <GenderFlower
    isVisible={true}
  />
</div>
`;

exports[`COMPONENT - Characters Robot renders confusion marks if robot is in default pose 1`] = `
.c0 {
  padding: 9.6rem 0 3rem 0;
  position: relative;
}

<div
  className="c0"
>
  <RobotArtwork
    pose="default"
  />
  <ConfusionMarks />
  <GenderFlower
    isVisible={false}
  />
</div>
`;

exports[`COMPONENT - Characters Robot renders correctly with default props 1`] = `
.c0 {
  padding: 9.6rem 0 3rem 0;
  position: relative;
}

<div
  className="c0"
>
  <RobotArtwork
    pose="default"
  />
  <ConfusionMarks />
  <GenderFlower
    isVisible={false}
  />
</div>
`;

exports[`COMPONENT - Characters Robot renders flower if female gender is passed 1`] = `
.c0 {
  padding: 9.6rem 0 3rem 0;
  position: relative;
}

<div
  className="c0"
>
  <RobotArtwork
    pose="default"
  />
  <ConfusionMarks />
  <GenderFlower
    isVisible={true}
  />
</div>
`;
